Notes/Strasnote.Notes.Data.Entities/EncryptedEntity.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Strasnote
+// Licensed under https://strasnote.com/licence
+
+using System;
+
+namespace Strasnote.Notes.Data.Entities
+{
+	/// <summary>
+	/// Enrypted entity
+	/// </summary>
+	public sealed record EncryptedEntity : IEntity
+	{
+		/// <inheritdoc/>
+		public long Id =>
+			EnryptedId;
+
+		/// <summary>
+		/// Enryption ID
+		/// </summary>
+		public long EnryptedId { get; init; }
+
+		/// <summary>
+		/// Encrypted Object (e.g. Folder / Note) ID
+		/// </summary>
+		public long EncryptedObjectId { get; init; }
+
+		/// <summary>
+		/// Encrypted Object Type (e.g. Folder / Note)
+		/// </summary>
+		public EncryptedObject EncryptedObjectType { get; init; }
+
+		/// <summary>
+		/// The Encryption Key for this object (encryted using this User's Public Key)
+		/// </summary>
+		public string EncryptedKeyHash { get; init; } = string.Empty;
+
+		/// <summary>
+		/// [Optional] Expiry date of this encryption key
+		/// </summary>
+		public DateTimeOffset? EncryptedExpiry { get; init; }
+
+		#region Relationships
+
+		/// <summary>
+		/// The ID of the User this Encrypted item belongs to
+		/// </summary>
+		public long UserId { get; init; }
+
+		#endregion
+
+		#region Lookups
+
+		/// <summary>
+		/// The User this Encrypted item belongs to
+		/// </summary>
+		public UserEntity? User { get; set; }
```

Comment:
I don't think a User should ever be nullable at this point as somebody has to own the encryption key

Reply:
So these lookups are all nullable because they aren't filled by the initial database query / map (as they aren't db columns) - they are optionally added later.  I thought they might be useful to be in there, but good to be nullable as they won't always be set?

Reply:
(UserId will always be set)

Reply:
Yep, agreed, that sounds good